tree/dataframe/inc/ROOT/RDFHelpers.hxx

```diff
@@ -181,6 +184,110 @@ RNode AsRNode(NodeType node)
 // clang-format on
 void RunGraphs(std::vector<RResultHandle> handles);
 
+
+// clang-format off
+/// RDF progress helper.
+/// This class provides callback functions to RDataFrame, which record event statistics
+/// and print them to the terminal every second.
+/// ProgressHelper::operator()(unsigned int, T&) is thread safe, and can be used as callback in MT mode.
+// clang-format on
+class ProgressHelper {
+public:
+
+  /// Create a progress helper.
+  /// \param increment RDF callbacks are called every `n` events. Pass this `n` here.
+  /// \param maxEvent If known, pass the number of events in the dataset. This will allow for computing completion
+  /// statistics.
+  /// \param progressBarWidth Number of characters the progress bar will occupy.
+  /// \param useShellColours Use shell colour codes to colour the output.
+  ProgressHelper(std::size_t increment,
+      std::size_t maxEvent,
+      unsigned int progressBarWidth = 40,
+      bool useShellColours = true) :
```

Comment:
Can you initialize this to `false` if Windows or `!isatty(fileno(stdout))`?

Reply:
Tell me how to detect that, and yes!
It's configurable because it should be, but I don't know how to auto-detect.

Also, we cannot use carriage return when writing to a file, so you see funny stuff in that file. I'd argue that it shouldn't be used when writing to files or we need a non-interactive mode.

Reply:
Can you first find out what happens on Windows, even with `useShellColours = false`?

Reply:
Oh, and we spell that as `useShellColors`, I guess in recognition of ROOT's first user collaborations, which were at Fermilab ;-)

Reply:
👍 for tty. How could we test this on windows? @bellenot

Reply:
@hageboeck I'll have to look at the implementation

Reply:
@hageboeck after a quick look at the code, I confirm that `useShellColours` must be `false` on Windows